### PR TITLE
Updated writeLevel to use the new nbt builder

### PR DESCRIPTION
**Pull request by GroobleDierne**
Currently `writeLevel()` only let you define the `RandomSeed` field but level.dat aims to contain way more data, you can now add whatever data you want. This PR is a breaking change (for flying-squid for instance) but it avoid introducing breaking changes in the future when there will be a need to add more data.

### CODE REVIEW

File: src/level.js [hunk 1]
```diff
@@ -19,22 +19,11 @@ async function readLevel (path) {
   return nbt.simplify(dnbt).Data
 }
 
-async function writeLevel (path, value) {
-  const nbt = {
+async function writeLevel (path, levelData) {
+  const data = await writeAsync({
```
Comment (u9g) [inline src/level.js]: can you make this an nbt.comp() with a name property?

Comment (GroobleDierne) [reply src/level.js]: Something like that ?
```javascript
const compound = nbt.comp({ Data: levelData })
compound.name = ''
const data = await writeAsync(compound)
```

Comment (u9g) [reply src/level.js]: yeah

**Comment by GroobleDierne** [comment]
By the way, https://github.com/PrismarineJS/prismarine-nbt/pull/48 is missing the implementation of NBTLong and NBTDouble.

**Comment by u9g** [comment]
> By the way, [PrismarineJS/prismarine-nbt#48](https://github.com/PrismarineJS/prismarine-nbt/pull/48) is missing the implementation of NBTLong and NBTDouble.

its missing a lot yeah, you can add that in this pr if its needed

**Comment by GroobleDierne** [comment]
Ready IMO

**Comment by u9g** [comment]
if it doesnt have any data anymore, whats the point in storing this function here, if you create the actual nbt data in user code

**Comment by GroobleDierne** [comment]
> 
> 
> if it doesnt have any data anymore, whats the point in storing this function here, if you create the actual nbt data in user code

I agree, not much of interest other than providing an abstraction layer to compress the data and write the file. It's just a utility function like readLevel so I think it makes sense to keep it.

**Comment by rom1504** [comment]
this is breaking and I don't see the benefit, please rethink it if you need the feature

**Pull request closed without merging**